CONTRIBUTING.md

```diff
@@ -1,136 +1,121 @@
+for our testing guidelines and overview of the testing infrastructure.
+2. Feel free to submit draft PRs to get early feedback and to make sure you are
+on the right track.
+3. A PR can contain any number of commits and when merged, the commits will be
+squashed into a single commit.
+4. The PR name should follow the template: `<type>: <name>`.  
+Where `type` is:
+   - `fix` for bug fixes;
+   - `feat` for new features;
+   - `refactor` for changes that reorganize code without adding new content;
+   - `doc` for changes that change documentation or comments;
+   - `test` for changes that introduce new tests;
+   - `chore` for grunt tasks like updating dependencies.
+5. The PR description should follow the following template:
 
 ```
-<type>: <name>
-
 <description>
 
 Test plan
 ---------
 <test plan>
 ```
+6. If your PR introduces a user-observable change (e.g. a new protocol
+feature, new configuration option, new Prometheus metric etc.) please
+document it in [CHANGELOG.md](CHANGELOG.md) in `[unreleased]` section.
 
-Where `type` is `fix` for fixes, `feat` for features, `refactor` for changes that primarily reorganize code, `doc` for changes that primarily change documentation or comments, and `test` for changes that primarily introduce new tests. The type is case sensitive.
-
-The `test plan` should describe in detail what tests are presented, and what cases they cover.
-
-If your PR introduces a new protocol feature, please document it in [CHANGELOG.md](CHANGELOG.md) under `unreleased`.
-
-
-### After the PR is submitted
-
-1. We have a CI process configured to run all the sanity tests on each PR. If the CI fails on your PR, you need to fix it before it will be reviewed.
-2. Once the CI passes, you should expect the first feedback to appear within one business day.  Please see the [code review process](#code-review-process) section for how we conduct code reviews.
-3. When all the comments have been addressed and all the reviewers are satisfied, they will approve your PR.  You can now add the `S-automerge` label on it to instruct Github to merge the PR.
-
-## Proposing new ideas and features
-
-If you want to propose an idea or a feature and work on it, create a new issue in the `nearcore` repository. We presently do not have an issue template.
+## After the PR is submitted
 
-You should expect someone to comment on the issue within 48 hours after it is created. If the proposal in the issue is accepted, you should then follow the process for `Working on current tasks` above.
+1. We have a CI process configured to run various tests on each PR.  All tests
+need to pass before your PR can be merged.
+2. When the CI is green and comments from the reviewer(s) have been addressed,
+they should approve the PR enabling you to merge it.
+3. You can use the `S-automerge` label to instruct Github to put the PR in the
+merge queue with other approved PRs so you do not have to manually look for a
+window to merge it.
```

Comment:
Let's rephrase this a bit maybe:

> To merge an approved PR, add `S-automerge` label. The label is added by PR's author if they have commit access, and by a reviewer otherwise. PR author can also apply automerge label immediaetly after filing a PR: this could remove one round-trip after PR is approved. 

Three bits of info: 

* it's on author to `automerge`
* except that if you don't have commit access
* document pre-emptive auto-merge trick

Reply:
Commit access doesn’t tie into it though, does it? Anyone can apply the label, don’t they? It’s just that what usually happens is if we have external contributors they don’t know about it so we end up applying the label.  I’d just change the last point to:

```suggestion
3. You can use the `S-automerge` label to put the PR onto the merge queue with
other approved PRs.
```

Reply:
>Commit access doesn’t tie into it though, does it?

I think it does: you need to have a triage permission to label issues and PR. Not exactly commit access, but I don't think an external contributor can label even their own PRs.

Reply:
Oh, I did not realise external folks cannot access labels.  I suppose it makes sense that they cannot do that.  I suppose if we had a system like [bors](https://github.com/bors-ng/bors-ng) that works via comments, then external folks should be able to merge their own PRs.  Not proposing that we switch but next time we are shopping, we should consider this issue.

I'll update the text.